### PR TITLE
Add entity_cache module to core #74

### PR DESCRIPTION
**Pull request by mglaman**
Picking up from #95 for backdrop/backdrop-issues#74

### CODE REVIEW

File: core/modules/entity/entitycache/entitycache.module [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+  }
+}
+
+/**
+ * Entity cache helper.
+ *
+ * Note: while this class is not a real entity controller it needs to extend
+ * DrupalDefaultEntityController to get access to protected properties.
+ */
+class EntityCacheControllerHelper extends DrupalDefaultEntityController {
```
Comment (mglaman) [inline core/modules/entity/entitycache/entitycache.module]:
```
Fatal error: Class 'DrupalDefaultEntityController' not found in /home/travis/build/backdrop/backdrop/core/modules/entity/entitycache/entitycache.module on line 25
ERROR: FATAL DatabaseSelectCloneTest: test runner returned a non-zero error code (255).
```

**Comment by Gormartsen** [comment]
@mglaman It will be good to replace Drupal with Backdrop as well. I see there is many tests provided by this module is failing as well.

**Comment by mglaman** [comment]
Yeah, I did a straight patch port from previous PR to just get started. First time playing around :)

**Comment by Gormartsen** [comment]
Ah, I see.
Have fun. Check our chat as well here: https://gitter.im/backdrop/backdrop-issues

File: core/modules/entity/entitycache/entitycache.test [hunk 2]
```diff
@@ -0,0 +1,1253 @@
+<?php
+module_load_include('test', 'book');
+
+/**
+ * Copy of BookTestCase.
+ */
+class EntityCacheBookTestCase extends BookTestCase {
```
Comment (mglaman) [inline core/modules/entity/entitycache/entitycache.test:9]:
```
Fatal error: Class 'BookTestCase' not found in /home/travis/build/backdrop/backdrop/core/modules/entity/entitycache/entitycache.test on line 7
```

**Comment by mglaman** [comment]
- [ ] Fix broken class references
- [ ] Remove license file, not needed since in core

**Pull request closed without merging**